operators/src/adapters/sparse_tiles_fill_adapter.rs

```diff
@@ -0,0 +1,381 @@
+                            && *this.current_time == tile.time =>
+                    {
+                        debug_assert!(this.next_tile.is_none());
+                        *this.state = State::PollingForNextTile;
+                        Some(Ok(tile))
+                    }
+                    // this is a new tile and we are at the start of a new grid run
+                    Some(Ok(tile))
+                        if *this.current_idx == tile.tile_position
+                            && *this.current_idx == min_idx =>
+                    {
+                        debug_assert!(this.next_tile.is_none());
+                        debug_assert!(tile.time.start() >= this.current_time.start());
+
+                        *this.current_time = tile.time;
+                        *this.state = State::PollingForNextTile;
+                        Some(Ok(tile))
+                    }
+                    // there is a new tile. Store it to use it in the next steps.
+                    Some(Ok(tile)) => {
+                        debug_assert!(this.next_tile.is_none());
+                        debug_assert!(tile.time.start() >= this.current_time.start());
+                        *this.next_tile = Some(tile);
+                        *this.state = State::FillAndProduceNextTile;
+                        Some(Ok(current_no_data_tile))
+                    }
+                    // an error ouccured, stop producing anything and return the error.
+                    Some(Err(e)) => {
+                        *this.state = State::Ended;
+                        return Poll::Ready(Some(Err(e)));
+                    }
+                    // the source is empty (now). Remember that.
+                    None => {
+                        if *this.current_idx == min_idx {
+                            // there was a tile and it flipped the next index to the first one. => we are done.
+                            *this.state = State::Ended;
+                            None
+                        } else if *this.current_idx == this.grid_bounds.max_index() {
+                            // this is the last tile
+                            *this.state = State::Ended;
+                            Some(Ok(current_no_data_tile))
+                        } else {
+                            // there was a tile and it was not the last one. => fill to end.
+                            *this.state = State::FillToEnd;
+                            Some(Ok(current_no_data_tile))
+                        }
+                    }
+                };
+                // move the current_idx. There is no need to do time progress here. Either a new tile sets that or it is not needed to fill to the end of the grid.
+                *this.current_idx = wrapped_next_idx;
+                Poll::Ready(res)
+            }
+            // the tile to produce is the the one stored
+            State::FillAndProduceNextTile if is_next_tile_stored => {
```

Comment:
don't you have to use the function with that name instead of the variable? otherwise it is always the same value

Reply:
i can't call the method since this is pinned and i have no access to the method here. This is why i call the method before calling project.  Since we can only fall into one case on each call the variable is always correct.